lib/routes/notifications.js

```diff
@@ -0,0 +1,156 @@
+const router = require('koa-router')()
+const models = require('../models')
+
+router
+  .get('/notifications', async (ctx) => {
+    ctx.body = await models.db.notification.findAll({
+      order: [['priority', 'ASC'], ['message_date', 'DESC']],
+      include: [{
+        model: models.db.participant,
+        through: {attributes: []}
+      }]
+    })
+  })
+  .get('/notifications/:id', async (ctx) => {
+    ctx.body = await models.db.notification.findById(
+      ctx.params.id, {
+        include: [{
+          model: models.db.participant,
+          through: {attributes: []}
+        }]
+      })
+  })
+  .post('/notifications', async (ctx) => {
+    let message = ctx.request.body.message
+    let messageDate = ctx.request.body.message_date
+    let expiryDate = ctx.request.body.expiry_date
+    let priority = ctx.request.body.priority
+    let eventId = ctx.request.body.event_id
+
+    if (!messageDate) {
+      messageDate = Date.now()
+    }
+    if (!eventId) {
+      eventId = 0
+    }
+
+    // Verify message with same date and eventId are not used to comply with uniqueness constraints
+    let notification = await models.db.notification.findOne({where: {[models.db.sequelize.Op.and]: [{message}, {message_date: messageDate}, {event_id: eventId}]}})
+    if (notification) {
+      ctx.status = 409
+      ctx.body = {'error': {
+        'code': 409,
+        'message': 'this notification message for message_date and event_id already exists'
+      }}
+    } else {
+      await models.db.notification
+        .findOrCreate({where: {
+          message: message,
+          message_date: messageDate,
+          expiry_date: expiryDate,
+          priority: priority,
+          event_id: eventId
+        }})
+        .spread(function (event, created) {
+          if (created) {
+            ctx.status = 201
+            ctx.body = event
+          } else {
+            ctx.status = 409
+            ctx.body = {'error': {
+              'code': 409,
+              'message': 'this notification message for message_date and event_id already exists'
+            }}
+          }
+        })
+    }
+  })
+
+  .patch('/notifications/:id', async (ctx) => {
+    let notificationId = ctx.params.id
+    let message = ctx.request.body.message
+    let messageDate = ctx.request.body.message_date
+    let eventId = ctx.request.body.event_id
+
+    if (!message && !messageDate && !eventId) {
```

Comment:
I think you maybe meant to use `||` here but that logic clashes somewhat with the following line where you search for conflicts, so unsure of how you meant to structure this check